test/user.test.ts

```diff
@@ -87,6 +87,34 @@ describe('Query user', () => {
     expect(error.code).to.be.equal(401);
     expect(error.additionalInfo).to.be.equal('Missing JWT token.');
   });
+
+  it('should return addresses if they exist', async () => {
+    const addresses = generateAddresses();
+
+    const userRepository = User.getRepository();
+    const firstUser = await userRepository.findOne({ email: 'first@user.com' });
+    const secondUser = await userRepository.findOne({ email: 'second@user.com' });
+    secondUser.addresses = addresses;
+
+    await userRepository.save(secondUser);
+
+    const firstUserToken = generateJwt(firstUser, false);
+    queryUser.variables.id = secondUser.id;
+
+    const response = await request('localhost:4000').post('/').send(queryUser).set('Authorization', firstUserToken);
+    const responseAddresses = response.body.data.user.addresses;
+
+    for (let i = 0; i < addresses.length; i++) {
+      expect(responseAddresses[i].cep).to.be.equal(addresses[i].cep);
+      expect(responseAddresses[i].street).to.be.equal(addresses[i].street);
+      expect(responseAddresses[i].streetNumber).to.be.equal(addresses[i].streetNumber);
+      expect(responseAddresses[i].neighborhood).to.be.equal(addresses[i].neighborhood);
+      expect(responseAddresses[i].city).to.be.equal(addresses[i].city);
+      expect(responseAddresses[i].state).to.be.equal(addresses[i].state);
+      expect(responseAddresses[i].complement).to.be.equal(addresses[i].complement);
+      expect(responseAddresses[i].id).to.be.equal(addresses[i].id);
```

Comment:
Reforçando a recomendação do `to.be.deep.eq`. Além da comparação strict ele tbm tem a vantagem de deixar o cóigo mais ilimpo.